.gitlab-ci.yml

```diff
@@ -0,0 +1,169 @@
+# I don't know what the minimum size is, but we cannot build on t3.micro.
+
+# TODO: Factor common builds between different tests.
+
+# The parameters for our job matrix:
+#
+# 1. Generator (Make, Ninja, MSBuild)
+# 2. Compiler (GCC, Clang, MSVC)
+# 3. Build type (Debug, Release)
+# 4. Definitions (-Dunity=OFF, -Dassert=ON, ...)
+
+
+.job_linux_build_test:
+  only:
+    variables:
+      - $CI_PROJECT_URL =~ /^https?:\/\/gitlab.com\//
+  stage: build
+  tags:
+    - linux
+    - c5.2xlarge
+  image: thejohnfreeman/rippled-build-ubuntu:develop
```

Comment:
how is this going to work when I want to test out builds with some other version of boost? or openssl ? I might have master that needs to build against a certain set and develop that needs to build against entirely different deps.

Reply:
I'm assuming that an individual Git tree is expecting to build against exactly one version of Boost or OpenSSL. The `.gitlab-ci.yml` in that tree should choose an image that has those versions. That means, in your example, that the `.gitlab-ci.yml` on `master` has one image and the `.gitlab-ci.yml` on `develop` has a different image. The developer is expected to update the `.gitlab-ci.yml` exactly when they change any other code to expect a new version. Does that make sense?

The way we tag the images on Docker Hub is up-in-the-air. We want immutable tags so that old builds can be repeated deterministically. We could just tag an image with a unique prefix of its ID. How does that sound?